doc/TypePal/RenameRefactoring/RenameRefactoring.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Rename refactoring
+---
+
+#### Synopsis
+
+TypePal offers a framework for rename refactoring. A `Renamer` collects document edits and diagnostics.
+
+#### Description
+
+The rename refactoring is one of the most commonly used refactorings; it renames all corresponding definitions and references to a new name. TypePal includes a framework that enables efficient implementation of rename refactoring for a language by removing boilerplate and generic default behaviour.
+
+::: Prerequisite
+The language uses a TypePal-based type-checker.
```

Comment:
Google says: The terms "type-checker" and "typechecker" refer to the same concept: a tool or mechanism that verifies the type correctness of code, either during compilation (static type checking) or runtime (dynamic type checking). While "type-checker" is the more common hyphenated form, "typechecker" is also a valid, widely used spelling. 

We usually write "typechecker".